### PR TITLE
feat: support remote JSON configs

### PR DESCRIPTION
**Pull request by Reecepbcups**
## Summary
Adds remote JSON configuration files to local-ic for start

## Why
To support spawn testnets on hetzner, gcp, etc private repos must also be supported. To accomplish this, the remote config can be public (gist), the docker build binary can (ghcr), but the codebase remain private

This way local-ic has everything it needs (docker image + config) to run, without exposing the private code

### CODE REVIEW

File: local-interchain/interchain/config.go [hunk 1]
```diff
@@ -45,14 +33,43 @@ func LoadConfig(installDir, chainCfgFile string) (*types.Config, error) {
...
 	log.Println("Using chain config:", cfgFilePath)
 
+	return setConfigDefaults(&config), nil
+}
+
+func LoadConfigFromURL(url string) (*types.Config, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		log.Fatalln(err)
+	}
```
Comment (jtieri) [inline local-interchain/interchain/config.go:56]: It is the callers responsibility to ensure the response body of an http request is closed after use, failure to do so can lead to resource leak, etc.

```suggestion
	resp, err := http.Get(url)
	if err != nil {
		log.Fatalln(err)
	}
	defer resp.Body.Close()
```

File: local-interchain/cmd/local-ic/start_chain.go [hunk 2]
```diff
@@ -40,6 +45,28 @@ var startCmd = &cobra.Command{
...
+		var err error
+
+		if strings.HasPrefix(configPath, "http") {
+			config, err = interchain.LoadConfigFromURL(configPath)
+			if err != nil {
+				panic(err)
+			}
+
+			// last part of the URL to be the test name
+			configPath = configPath[strings.LastIndex(configPath, "/")+1:]
```
Comment (jtieri) [inline local-interchain/cmd/local-ic/start_chain.go:58]: `configPath` can become any arbitrary string depending on what URL the user passes in, is this okay? is it solely used for creating the name of the running test case?

**Review by jtieri — COMMENTED** [review]
just had the one suggestion otherwise this looks good

File: local-interchain/interchain/config.go (hunk 1 above)
Comment (Reecepbcups) [reply local-interchain/interchain/config.go:56]: great catch

File: local-interchain/cmd/local-ic/start_chain.go (hunk 2 above)
Comment (Reecepbcups) [reply local-interchain/cmd/local-ic/start_chain.go:58]: yes, it is only used as the test name (and technically nothing here is also okay as other data is appended)

else you get a super long testname for gist which takes up the an entire terminal line. really messy

**Review by jtieri — APPROVED** [review]

**Pull request merged**